src/test/osd/test_scrubber_be.cc

```diff
@@ -0,0 +1,870 @@
+
+
+// note: the actual owner of the OSD "objects" that are used by
+// the mockers
+class TestTScrubberBe : public ::testing::Test {
+ public:
+  // the test data source
+
+  virtual TestTScrubberBeParams inject_params() = 0;
+
+  // initial test data
+
+  // CephContext* cct{nullptr};
+  // LogChannelRef logger;
+  ScrubGenerator::MockLog logger;
+  ScrubGenerator::pool_conf_t pool_conf;
+  ScrubGenerator::RealObjsConf real_objs;
+  int num_osds{0};
+
+
+  // ctor & initialization
+
+  TestTScrubberBe() = default;
+
+  ~TestTScrubberBe() = default;
+
+  void SetUp() override
+  {
+    std::cout << "TestTScrubberBe::SetUp()\n";
+    logger.err_count = 0;
+
+    // fetch test configuration
+    auto params = inject_params();
+    pool_conf = params.pool_conf;
+    real_objs = params.objs_conf;
+    // logger = params.logger;
+    num_osds = params.num_osds;
+
+    // create the OSDMap
+
+    osdmap = setup_map(num_osds, pool_conf);
+
+    std::cout << "osdmap: " << *osdmap << std::endl;
+
+    // extract the pool from the osdmap
+
+    pool_id = osdmap->lookup_pg_pool_name(pool_conf.name);
+    const pg_pool_t* ext_pool_info = osdmap->get_pg_pool(pool_id);
+    pool =
+      std::make_shared<PGPool>(osdmap, pool_id, *ext_pool_info, pool_conf.name);
+
+    std::cout << "pool: " << pool->info << std::endl;
+
+    // a PG in that pool?
+    info = setup_pg_in_map();
+    // std::cout << fmt::format("info: {}\n", info);
+    std::cout << "info: " << info << std::endl;
+
+
+    real_objs_list =
+      ScrubGenerator::make_real_objs_conf(pool_id, real_objs, acting_osds);
+
+    // now we can create the main mockers
+
+    // the "PgScrubber"
+    test_scrubber = std::make_unique<TestScrubber>(spg, osdmap, &logger);
+    std::cout << "test_scrubber: "
+              << "test scrubber created" /* *test_scrubber*/ << std::endl;
+
+    // the "Pg" (and its backend)
+    test_pg = std::make_unique<TestPg>(pool, info, i_am);
+    std::cout << fmt::format("{}: test_pg: OK\n", __func__);
+    std::cout << fmt::format("{}: acting: {}\n", __func__, acting_shards);
+    sbe = std::make_unique<TestScrubBackend>(*test_scrubber,
+                                             *test_pg,
+                                             i_am,
+                                             /* repair? */ false,
+                                             scrub_level_t::deep,
+                                             acting_shards);
+
+    // create a osd-num only copy of the relevant OSDs
+    acting_osds.reserve(acting_shards.size());
+    for (const auto& shard : acting_shards) {
+      acting_osds.push_back(shard.osd);
+    }
+
+    sbe->new_chunk();
+    std::cout << fmt::format("{}: TestScrubBackend created\n", __func__);
+    fake_a_scrub_set(real_objs_list);
+    std::cout << fmt::format("{}: fake_a_scrub_set() done\n", __func__);
+  }
+
+  void TearDown() override;
+
+  // void fake_a_scrub_set(ScrubGenerator::RealObjsConf conf);
```

Comment:
Instead of this comment, indicate what the function actually does.

Reply:
done